12-watchtower-API.md

```diff
@@ -4,245 +4,337 @@
 
-Due to replace-by-revocation lightning channels, the client should send data to the WatchTower for every new update in the channel, otherwise the WatchTower may not be able to respond to specific breaches. 
+Due to replace-by-revocation Lightning channels, the client should send data to the WatchTower for every new update in the channel, otherwise the WatchTower may not be able to respond to specific breaches. 
 
 Finally, optional QoS can be offered by the WatchTower to provide stronger guarantees to the client, such as a signed receipt for every new job. The rationale for the receipt is to build an _accountable_ WatchTower as the customer can later use it as publicly verifiable evidence if the WatchTower fails to protect them.
 
 The scope of this document includes: 
 
-- A protocol for client/server comunication.
-- How to build appointments for the WatchTower, inluding key/locator derivation and data encryption.
+- A protocol for client/server communication.
+- How to build appointments for the WatchTower, including key/locator derivation and data encryption.
 - A format for the signed receipt. 
 
 The scope of this bolt does not include: 
 
  - A payment protocol between the customer and WatchTower. 
  - WatchTower server discovery.
  
+For the rest of this document we will refer to the WatchTower as server, and the user/Ligthning node as client.
 
-## Table of Contents
+## Table of Contents 
+* [WatchTower discovery](#watchtower-discovery)
+* [WatchTower services](#watchtower-discovery)
+	* [Basic Service](#basic-service)
+	* [Quality of Service](#quality-of-service)
+* [Sending and receiving appointments](#sending-and-receiving-appointments)
+ 	* [The `appointment` message](#the-appointment-message)
+ 	* [The `appointment_accepted` message](#the-appointment_accepted-message)
+ 	* [The `appointment_rejected` message](#the-appointment_rejected-message)
+* [Quality of Service data](#quality-of-service-data)
+	* [`accountability`](#accountability)
+* [Transaction Locator and Encryption Key](#transaction-locator-and-encryption-key)
+* [Encryption Algorithms and Parameters](#encryption-algorithms-and-parameters)
+* [Payment Modes](#payment-modes)
+* [No compression of justice transaction](#no-compression-of-justice-transaction)
 
-  * [Transaction Locator and Encryption Key](#transaction-locator-and-encryption-key)
-  * [Encryption Algorithms and Parameters](#encryption-algorithms-and-parameters)
-  * [WatchTower API](#WatchTower-api)
-  * [Signed Receipt Fields](#signed-receipt-fields)
-  * [Payment Modes](#payment-modes)
-  * [Number of updates](#number-of-updates)
-  * [No compression of justice transaction](#no-compression-of-justice-transaction)
+## WatchTower discovery
+At this point we're leaving the client/server connection to be protocol agnostic. How the Lightning node finds the WatchTower or how the WatchTower announces their presence and services provided is not specified.
```

Comment:
The english is a bit off. Let me fix it: 

```
We have not defined how a client can find a list of servers to hire. We assume the client has found a server and the server is offering a quality of service. As well, the client has an auth_token that the server will recognise when accepting the job (e.g. an unblinded pre-paid token). 

```

Reply:
I think that's way too specific. The server does't have to be offering any `qos` and the client is not always required to provide a token.

Reply:
Feel free to rephrase it if you think the english is off though, but it's a general idea without entering in every single case, in the same way that the second paragraph of the overview section gives some notes about what the protocol is about.

Reply:
We have not defined how a client can find a list of servers to hire. We assume the client has found a server and the server is offering a watching service. The service can either be the basic service or an accountable quality of service. To deal with pre-payments (when necessary), the client may have an 'auth_token') that the server can verify when accepting the job (e.g. an unblinded token). 

better?

Reply:
I like this better, yeah.

I guess you mean blinded instead of unblinded tho.

Reply:
In this section 'WatchTower Discovery' there are now three paragraphs. You can delete the first two, and then slightly modify the third to fix it (i.e. with qos or auth_token). They are both saying the same thing.